packages/kbn-dev-utils/src/plugins/parse_kibana_platform_plugin.ts

```diff
@@ -23,20 +23,27 @@ import loadJsonFile from 'load-json-file';
 export interface KibanaPlatformPlugin {
   readonly directory: string;
   readonly manifestPath: string;
-  readonly manifest: {
-    id: string;
-    ui: boolean;
-    server: boolean;
-    [key: string]: unknown;
-  };
+  readonly manifest: Manifest;
+}
+
+interface Manifest {
+  id: string;
+  ui: boolean;
+  server: boolean;
+  kibanaVersion: string;
+  version: string;
+  requiredPlugins: readonly string[];
+  optionalPlugins: readonly string[];
+  requiredBundles: readonly string[];
+  extraPublicDirs: readonly string[];
 }
 
 export function parseKibanaPlatformPlugin(manifestPath: string): KibanaPlatformPlugin {
   if (!Path.isAbsolute(manifestPath)) {
     throw new TypeError('expected new platform manifest path to be absolute');
   }
 
-  const manifest = loadJsonFile.sync(manifestPath);
+  const manifest: Partial<Manifest> = loadJsonFile.sync(manifestPath);
```

Comment:
Platform validates `manifest` file https://github.com/elastic/kibana/blob/4c236696119a248f0650cd6a19578d8236f611a1/src/core/server/plugins/discovery/plugin_manifest_parser.ts
What do you think if we move this validation here? @spalger

Reply:
I'd like to avoid directly depending on src/core, and I don't think we need to worry about the duplicate validation, but I'm not opposed to a sharing of some sort. Ultimately we only validate the properties we use so sharing doesn't really buy us much IMO